### PR TITLE
Unit checking for mapbase metadata.

### PR DESCRIPTION
**Pull request by MSKirk**
Added unit checking for metadata keywords CUNIT and WAVEUNIT to show a warning if the units are not defined in astropy.units to resolve issue #1641.

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -566,15 +566,25 @@ def _get_cmap_name(self):
...
+
+        Allows for default unit assignment for:
+            CUNIT1, CUNIT2, WAVEUNIT,  """
 #        if (self.dsun <= 0 or self.dsun >= 40 * constants.au):
 #            raise InvalidHeaderInformation("Invalid value for DSUN")
+
+        warnings.simplefilter('always', Warning)
+
+        for meta_property in ('cunit1', 'cunit2', 'waveunit'):
+            if (self.meta.get(meta_property) != None) and (u.Unit(self.meta.get(meta_property), parse_strict = 'silent').physical_type == 'unknown'):
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: you do not need the `!= None` here, just `if self.meta.get(meta_property) and ...` will work. (None evaluates to False)

**Comment by Cadair** [comment]
So this will throw a warning, until the user tries to use it, when it will throw an exception?

File: sunpy/map/mapbase.py (hunk 1 above)
Comment (dpshelio) [inline sunpy/map/mapbase.py]: `!=` changed to `is not`

**Comment by MSKirk** [comment]
An exception is thrown when the user tries to do anything that requires a units to be evaluated by astropy.units (almost anything else).

**Comment by Cadair** [comment]
@MSKirk yes, cool.

**Comment by Cadair** [comment]
@MSKirk Could you add some  tests and a changelog entry please?

**Comment by MSKirk** [comment]
@Cadair Working on it now.

**Comment by Cadair** [comment]
Thanks!

**Comment by dpshelio** [comment]
Looks good to me :+1:

**Comment by Cadair** [comment]
:+1:

**Comment by wafels** [comment]
+1

**Comment by Cadair** [comment]
@MSKirk Can you merge master into this PR please? We have done some changes to the testing.

**Comment by MSKirk** [comment]
@Cadair Just merged master - still having a sphinx documentation build error.

**Comment by wafels** [comment]
The error log is weird...  The deprecation warnings at the end get repeated ad inifinitum.

reading sources... [ 95%] guide/data_types/lightcurve

reading sources... [ 95%] guide/data_types/maps

/home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-2.7/sunpy/map/mapbase.py:1482: Warning: This map is not properly oriented. Plot axes may be incorrect

  Warning)

reading sources... [ 96%] guide/data_types/spectra

/home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-2.7/sunpy/spectra/spectrogram.py:136: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future

  for n, mid in enumerate(self.midpoints[min_mid:]):

/home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-2.7/sunpy/spectra/spectrogram.py:598: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future

  return self.data[key]

/home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-2.7/sunpy/spectra/spectrogram.py:136: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future

  for n, mid in enumerate(self.midpoints[min_mid:]):

/home/travis/build/sunpy/sunpy/build/lib.linux-x86_64-2.7/sunpy/spectra/spectrogram.py:598: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future

**Comment by wafels** [comment]
Could you pull master on to your PR again to trigger another CI test?

Thanks!

**Comment by Cadair** [comment]
I have restarted the sphinx build.

**Comment by Cadair** [comment]
It's hit the error with all the builds at the moment, which should be fixed by #1629

**Comment by Cadair** [comment]
@MSKirk can you merge master into this now please? We have updated the CI configuration.

**Comment by MSKirk** [comment]
@Cadair Still an issue with CI.

**Comment by wafels** [comment]
@SolarBoy - we recently (as in the past few days) yet another continuous
integration check.  Quantified Code has detected that there is a docstring
missing in test_mapbase.  You could fix this just by putting in a short
docstring that gives a short description of what the test does.

**Comment by wafels** [comment]
Weird - that infinitely looping error stream I reported earlier is still around in the Travis-CI....

**Comment by MSKirk** [comment]
To quote the rock-philosopher Meatloaf: "Two out of three ain't bad."

**Pull request closed without merging**